app/domain/authentication/authn_jwt/jwt_fetch_jwks_uri_signing_key.rb

```diff
@@ -0,0 +1,51 @@
+require 'uri'
+require 'net/http'
+
+module Authentication
+  module AuthnJwt
+
+    JwtFetchJwksUriSigningKey = CommandClass.new(
+      dependencies: {
+        logger: Rails.logger,
+        discover_identity_provider: OAuth::DiscoverIdentityProvider.new,
+        fetch_secrets: Conjur::FetchRequiredSecrets.new
+      },
+      inputs: %i[authenticator_input]
+    ) do
+      def call
+        jwks_uri
+        fetch_jwks_keys
+      end
+
+      private
+
+      def jwks_uri
+        @logger.debug(LogMessages::Authentication::AuthnJwt::JwksUriResourceNameConfiguration.new(resource_id))
+        @jwks_uri_secret ||= @fetch_secrets.(resource_ids: [resource_id])
+        @jwks_uri ||=@jwks_uri_secret[resource_id]
+      end
```

Comment:
please change this to the following. Some points to see here:
1. You shouldn't have 2 memoized objects in one method. If you have that, then you can probably split the method into 2.
2. the call method needs to tell the story in english. what does `jwks_uri` mean? the action `fetch_jwks_uri` explains to the reader what is going on (and hinting that we are going to the DB to get it)
3. The log message also hinted that there's an "action method" to have here.

```suggestion
 def fetch_jwks_uri
  @logger.debug(LogMessages::Authentication::AuthnJwt::JwksUriResourceNameConfiguration.new(resource_id))
  jwks_uri
end

def jwks_uri
  @jwks_uri ||= jwks_uri_secret[resource_id]
end

def jwks_uri_secret
  @jwks_uri_secret ||= @fetch_secrets.(resource_ids: [resource_id])
end
```